### ISSUE TITLE
AKS - kubenet issues with Kubernetes 1.10.x

### ISSUE THREAD

**Issue by paulbouwer** [Azure/AKS#531]
### Is this an ISSUE or FEATURE REQUEST?

ISSUE

### Relevant information:

- Kubernetes 1.10.3, 1.10.5
- Service Principal used in configurations has Contributor role at Subscription scope.
- Azure CLI version is 2.0.41
- RBAC enabled for all configurations
- No monitoring enabled (via portal configurations)
- No http application routing enabled (except where explicitly mentioned)
- Multiple regions (Australia East, West Europe)
- Both kubenet and azure cni configurations work for Kubernetes 1.9.9

### What happened:

When deploying AKS cluster with Kubernetes 1.10.x and basic networking (kubenet), a number of pods are in `CrashLoopBackOff`. These are `azureproxy`, `kube-svc-redirect` and `kubernetes-dashboard`. It seems to be network related.

Deploying an AKS cluster with Kubernetes 1.10.x and advanced networking (azure cni), results in all the above mentioned pods `Running`.

### How to reproduce it:

**paul-aks-01 (azure cli, k8s 1.10.5)**

```bash
$ az aks create --name paul-aks-01 --resource-group Paul-AKS-01 --kubernetes-version 1.10.5 --location AustraliaEast --node-count 3 --node-vm-size "Standard_DS3_v2" --ssh-key-value "~/.ssh/id_rsa.pub" --service-principal "xxx" --client-secret "xxx"
```

```bash
$ k get pods --all-namespaces
NAMESPACE     NAME                                   READY     STATUS             RESTARTS   AGE
kube-system   azureproxy-6496d6f4c6-p7ngr            0/1       CrashLoopBackOff   30         2h
kube-system   heapster-5f7c7df649-g7pg7              2/2       Running            0          2h
kube-system   kube-dns-v20-58bc8dcd9f-5sz9t          3/3       Running            0          2h
kube-system   kube-dns-v20-58bc8dcd9f-dtt96          3/3       Running            0          2h
kube-system   kube-proxy-ff4bq                       1/1       Running            0          2h
kube-system   kube-proxy-nm567                       1/1       Running            0          2h
kube-system   kube-proxy-r7zlk                       1/1       Running            0          2h
kube-system   kube-svc-redirect-7g9sw                0/1       CrashLoopBackOff   21         2h
kube-system   kube-svc-redirect-l9vfp                0/1       CrashLoopBackOff   21         2h
kube-system   kube-svc-redirect-sfmv8                0/1       CrashLoopBackOff   21         2h
kube-system   kubernetes-dashboard-d876d4cdf-r2h5d   0/1       CrashLoopBackOff   29         2h
kube-system   tunnelfront-66478c7676-qm9z9           1/1       Running            12         2h
```

**paul-aks-02 (azure cli, k8s 1.10.3)**

```bash
$ az aks create --name paul-aks-02 --resource-group Paul-AKS-02 --kubernetes-version 1.10.3 --location AustraliaEast --node-count 3 --node-vm-size "Standard_DS3_v2" --ssh-key-value "~/.ssh/id_rsa.pub" --service-principal "xxx" --client-secret "xxx"
```

```bash
$ k get pods --all-namespaces
NAMESPACE     NAME                                   READY     STATUS             RESTARTS   AGE
kube-system   azureproxy-6496d6f4c6-qgnqh            0/1       CrashLoopBackOff   27         2h
kube-system   heapster-5f7c7df649-7j6cq              2/2       Running            0          2h
kube-system   kube-dns-v20-58bc8dcd9f-45s5c          3/3       Running            0          2h
kube-system   kube-dns-v20-58bc8dcd9f-6lxpk          3/3       Running            0          2h
kube-system   kube-proxy-4hx27                       1/1       Running            0          2h
kube-system   kube-proxy-v4vll                       1/1       Running            0          2h
kube-system   kube-proxy-x89v7                       1/1       Running            0          2h
kube-system   kube-svc-redirect-2ffs7                0/1       CrashLoopBackOff   19         2h
kube-system   kube-svc-redirect-6pk8n                0/1       CrashLoopBackOff   19         2h
kube-system   kube-svc-redirect-kdj4t                1/1       Running            19         2h
kube-system   kubernetes-dashboard-d876d4cdf-r59ps   0/1       CrashLoopBackOff   28         2h
kube-system   tunnelfront-8dd5bd855-k4v7w            1/1       Running            1          2h
```

**paul-aks-03 (azure cli, k8s 1.10.5, http application routing)**

```bash
$ az aks create --name paul-aks-03 --resource-group Paul-AKS-03 --kubernetes-version 1.10.5 --location AustraliaEast --node-count 3 --node-vm-size "Standard_DS3_v2" --enable-addons http_application_routing ---ssh-key-value "~/.ssh/id_rsa.pub" --service-principal "xxx" --client-secret "xxx"
```

```bash
$ k get pods --all-namespaces
NAMESPACE     NAME                                                              READY     STATUS             RESTARTS   AGE
kube-system   addon-http-application-routing-default-http-backend-b8f5bc6xznd   1/1       Running            0          2h
kube-system   addon-http-application-routing-external-dns-7b67bcd5b7-2s557      1/1       Running            0          2h
kube-system   addon-http-application-routing-nginx-ingress-controller-86jh56p   0/1       CrashLoopBackOff   43         2h
kube-system   azureproxy-6496d6f4c6-jmkw9                                       0/1       CrashLoopBackOff   27         2h
kube-system   heapster-5f7c7df649-b2522                                         2/2       Running            0          2h
kube-system   kube-dns-v20-58bc8dcd9f-f4ws5                                     3/3       Running            0          2h
kube-system   kube-dns-v20-58bc8dcd9f-v64g2                                     3/3       Running            0          2h
kube-system   kube-proxy-d5r9t                                                  1/1       Running            0          2h
kube-system   kube-proxy-fqltn                                                  1/1       Running            0          2h
kube-system   kube-proxy-xhfth                                                  1/1       Running            0          2h
kube-system   kube-svc-redirect-bc6fs                                           0/1       CrashLoopBackOff   19         2h
kube-system   kube-svc-redirect-cppr9                                           1/1       Running            19         2h
kube-system   kube-svc-redirect-sjpb9                                           0/1       CrashLoopBackOff   19         2h
kube-system   kubernetes-dashboard-d876d4cdf-q79kx                              0/1       CrashLoopBackOff   28         2h
kube-system   tunnelfront-79ddb75f6f-fznw9                                      1/1       Running            0          2h
```

```bash
$ k logs addon-http-application-routing-nginx-ingress-controller-86jh56p -n kube-system
-------------------------------------------------------------------------------
NGINX Ingress controller
  Release:    0.13.0
  Build:      git-4bc943a
  Repository: https://github.com/kubernetes/ingress-nginx
-------------------------------------------------------------------------------

I0716 05:11:59.135623       6 flags.go:162] Watching for ingress class: addon-http-application-routing
W0716 05:11:59.135664       6 flags.go:165] only Ingress with class "addon-http-application-routing" will be processed by this ingress controller
W0716 05:11:59.135979       6 client_config.go:533] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0716 05:11:59.136122       6 main.go:181] Creating API client for https://10.0.0.1:443
```

```bash
$ k logs azureproxy-6496d6f4c6-jmkw9 -n kube-system
2018/07/16 05:12:43 [emerg] 1#1: host not found in upstream "paul-aks--paul-aks-03-538326-43cf897e.hcp.australiaeast.azmk8s.io:443" in /etc/nginx/conf.d/default.conf:2
nginx: [emerg] host not found in upstream "paul-aks--paul-aks-03-538326-43cf897e.hcp.australiaeast.azmk8s.io:443" in /etc/nginx/conf.d/default.conf:2
```

```bash
$ az aks show -g Paul-AKS-03 -n paul-aks-03 --query "fqdn"
"paul-aks--paul-aks-03-538326-43cf897e.hcp.australiaeast.azmk8s.io"
```

**paul-aks-04 (portal, k8s 1.10.5, azure CNI - advanced networking)**

```bash
$ k get pods --all-namespaces
NAMESPACE     NAME                                   READY     STATUS    RESTARTS   AGE
kube-system   azureproxy-6496d6f4c6-ghknz            1/1       Running   2          1h
kube-system   heapster-5f7c7df649-s4rv7              2/2       Running   0          1h
kube-system   kube-dns-v20-58bc8dcd9f-ktn6b          3/3       Running   0          1h
kube-system   kube-dns-v20-58bc8dcd9f-n5858          3/3       Running   0          1h
kube-system   kube-proxy-j6wxc                       1/1       Running   0          1h
kube-system   kube-proxy-kg8fw                       1/1       Running   0          1h
kube-system   kube-proxy-pnbdm                       1/1       Running   0          1h
kube-system   kube-svc-redirect-8c5ws                1/1       Running   0          1h
kube-system   kube-svc-redirect-rkt4t                1/1       Running   0          1h
kube-system   kube-svc-redirect-vtvkz                1/1       Running   0          1h
kube-system   kubernetes-dashboard-d876d4cdf-vqqgg   1/1       Running   2          1h
kube-system   tunnelfront-698cdb9d9f-85jk5           1/1       Running   0          1h
```

**paul-aks-05 (portal, k8s 1.10.5, kubenet - basic networking)**

```bash
$ k get pods --all-namespaces
NAMESPACE     NAME                                   READY     STATUS             RESTARTS   AGE
kube-system   azureproxy-6496d6f4c6-c4xsg            0/1       CrashLoopBackOff   19         1h
kube-system   heapster-5f7c7df649-vg4kp              2/2       Running            0          1h
kube-system   kube-dns-v20-58bc8dcd9f-cv4nq          3/3       Running            0          1h
kube-system   kube-dns-v20-58bc8dcd9f-qndkj          3/3       Running            0          1h
kube-system   kube-proxy-9pn9g                       1/1       Running            0          1h
kube-system   kube-proxy-hgxjk                       1/1       Running            0          1h
kube-system   kube-proxy-vp2k9                       1/1       Running            0          1h
kube-system   kube-svc-redirect-rw5p6                0/1       CrashLoopBackOff   13         1h
kube-system   kube-svc-redirect-tm4mm                0/1       CrashLoopBackOff   13         1h
kube-system   kube-svc-redirect-zkdbr                0/1       CrashLoopBackOff   13         1h
kube-system   kubernetes-dashboard-d876d4cdf-6fnrd   0/1       CrashLoopBackOff   20         1h
kube-system   tunnelfront-dd8845ccd-vlk6f            1/1       Running            0          1h
```

```bash
$ k logs azureproxy-6496d6f4c6-c4xsg -n kube-system
2018/07/16 05:23:07 [emerg] 1#1: host not found in upstream "paul-aks-05-ee0da9a3.hcp.australiaeast.azmk8s.io:443" in /etc/nginx/conf.d/default.conf:2
nginx: [emerg] host not found in upstream "paul-aks-05-ee0da9a3.hcp.australiaeast.azmk8s.io:443" in /etc/nginx/conf.d/default.conf:2
```

```bash
$ az aks show -g Paul-AKS-05 -n paul-aks-05 --query "fqdn"
"paul-aks-05-ee0da9a3.hcp.australiaeast.azmk8s.io"
```

```bash
$ k logs kubernetes-dashboard-d876d4cdf-6fnrd -n kube-system
2018/07/16 05:31:58 Using in-cluster config to connect to apiserver
2018/07/16 05:31:58 Using service account token for csrf signing
2018/07/16 05:31:58 No request provided. Skipping authorization
2018/07/16 05:31:58 Starting overwatch
2018/07/16 05:32:28 Error while initializing connection to Kubernetes apiserver. This most likely means that the cluster is misconfigured (e.g., it has invalid apiserver certificates or service accounts configuration) or the --apiserver-host param points to a server that does not exist. Reason: Get https://10.0.0.1:443/version: dial tcp 10.0.0.1:443: i/o timeout
Refer to our FAQ and wiki pages for more information: https://github.com/kubernetes/dashboard/wiki/FAQ
```

```bash
$ k logs kube-svc-redirect-zkdbr -n kube-system
[ 05:31:25 ] ERR: APISERVER_FQDN is not defined, will check for redirector proxy
[ 05:31:25 ] INF: Starting: PROXY:kube-system:azureproxy KUBERNETES IP:10.0.0.1
[ 05:31:25 ] INF: API server ip was:
[ 05:31:25 ] INF: Proxy pod ip was:
[ 05:31:26 ] INF: Proxy pod ip now:
[ 05:31:26 ] ERR: Proxy POD IP is not available, attempt 1/12
[ 05:31:36 ] INF: Proxy pod ip was:
[ 05:31:36 ] INF: Proxy pod ip now:
[ 05:31:36 ] ERR: Proxy POD IP is not available, attempt 2/12
[ 05:31:46 ] INF: Proxy pod ip was:
[ 05:31:47 ] INF: Proxy pod ip now:
[ 05:31:47 ] ERR: Proxy POD IP is not available, attempt 3/12
[ 05:31:57 ] INF: Proxy pod ip was:
[ 05:31:57 ] INF: Proxy pod ip now:
[ 05:31:57 ] ERR: Proxy POD IP is not available, attempt 4/12
[ 05:32:07 ] INF: Proxy pod ip was:
[ 05:32:07 ] INF: Proxy pod ip now:
[ 05:32:07 ] ERR: Proxy POD IP is not available, attempt 5/12
[ 05:32:17 ] INF: Proxy pod ip was:
[ 05:32:17 ] INF: Proxy pod ip now:
[ 05:32:17 ] ERR: Proxy POD IP is not available, attempt 6/12
[ 05:32:27 ] INF: Proxy pod ip was:
[ 05:32:27 ] INF: Proxy pod ip now:
[ 05:32:27 ] ERR: Proxy POD IP is not available, attempt 7/12
[ 05:32:37 ] INF: Proxy pod ip was:
[ 05:32:38 ] INF: Proxy pod ip now:
[ 05:32:38 ] ERR: Proxy POD IP is not available, attempt 8/12
[ 05:32:48 ] INF: Proxy pod ip was:
[ 05:32:48 ] INF: Proxy pod ip now:
[ 05:32:48 ] ERR: Proxy POD IP is not available, attempt 9/12
[ 05:32:58 ] INF: Proxy pod ip was:
[ 05:32:58 ] INF: Proxy pod ip now:
[ 05:32:58 ] ERR: Proxy POD IP is not available, attempt 10/12
[ 05:33:08 ] INF: Proxy pod ip was:
[ 05:33:08 ] INF: Proxy pod ip now:
[ 05:33:08 ] ERR: Proxy POD IP is not available, attempt 11/12
[ 05:33:18 ] INF: Proxy pod ip was:
[ 05:33:18 ] INF: Proxy pod ip now:
[ 05:33:18 ] ERR: Proxy POD IP is not available, attempt 12/12
[ 05:33:28 ] ERR: EXIT as Proxy POD IP is not available, total attempts: 13
```

**paul-aks-06 (azure cli, k8s 1.10.5, azure CNI)**

Had to create a custom VNET and Subnet and attach to AKS create, otherwise got the following error from the az aks create command (in debug mode):

```json
{
  "code": "CidrStringParseError",
  "message": "Cidr  is invalid",
  "target": "networkProfile.serviceCIDR"
}
```

```bash
$ az network vnet create --name paul-aks-06 --resource-group Paul-AKS-06 --address-prefixes "10.0.0.0/8" --location AustraliaEast --subnet-name aks-subnet --subnet-prefix "10.240.0.0/16"

$ VNET_SUBNET_ID=$(az network vnet subnet show -g paul-aks-06 -n aks-subnet --vnet-name paul-aks-06 --query "id" -o tsv)

$ az aks create --name paul-aks-06 --resource-group Paul-AKS-06 --kubernetes-version 1.10.5 --location AustraliaEast --node-count 3 --node-vm-size "Standard_DS3_v2" --network-plugin azure --service-cidr "10.0.0.0/16" --dns-service-ip "10.0.0.10" --docker-bridge-address "172.17.0.1/16" --ssh-key-value "~/.ssh/id_rsa.pub" --service-principal "xxx" --client-secret "xxx" --vnet-subnet-id "$VNET_SUBNET_ID"
```

```bash
$ k get pods --all-namespaces
NAMESPACE     NAME                                   READY     STATUS    RESTARTS   AGE
kube-system   azureproxy-6496d6f4c6-vclrb            1/1       Running   2          39m
kube-system   heapster-5f7c7df649-q6nhb              2/2       Running   0          39m
kube-system   kube-dns-v20-58bc8dcd9f-dbfs8          3/3       Running   0          39m
kube-system   kube-dns-v20-58bc8dcd9f-f9gtc          3/3       Running   0          39m
kube-system   kube-proxy-45vw9                       1/1       Running   0          39m
kube-system   kube-proxy-q7q7k                       1/1       Running   0          39m
kube-system   kube-proxy-z8v7w                       1/1       Running   0          39m
kube-system   kube-svc-redirect-kspx7                1/1       Running   0          39m
kube-system   kube-svc-redirect-sdzbs                1/1       Running   0          39m
kube-system   kube-svc-redirect-vp7pr                1/1       Running   0          39m
kube-system   kubernetes-dashboard-d876d4cdf-vx8vl   1/1       Running   2          39m
kube-system   tunnelfront-9db97f66c-btwqr            1/1       Running   0          39m
```

**Comment by TiagoJMartins** [comment]
Experiencing the same issue on westeurope. Tried 1.10.5 and 1.9.9.
azureproxy's nginx can't resolve the upstream kubernetes api endpoint.

**Comment by mbrancato** [comment]
hey @paulbouwer , can you run some of your `kubectl get pods` commands with the `-o wide` option to show their IP and node? I think you're having the similar issues we've been experiencing in #533.

**Comment by neoakris** [comment]
(made an account to post this)
I ran into this issue as well. 
It turned out that if you use the default kubenet networking and deploy to a prexisting vnet then you get those crashes and kubectl logs doesn't work. (Somewhere on their website it says that kubenet networking is meant to be 100% microsoft managed, and if you read between the lines it means that kubenet networking isn't supported when deployed to a preexisting network, but the CLI will still let you do it/the cli verification phase doesn't stop you from doing a known unsupported operation. I reported it as a bug to a MS engineer recently.)

Only azure-cni lets you deploy to a preexisting vnet.
